.github/workflows/master-push.yml

```diff
@@ -86,10 +86,30 @@ jobs:
         echo 'from setuptools import setup; setup(name="gdcm.py")' > gdcmbin/bin/setup.py
         python -m pip install -e gdcmbin/bin/
         python -c 'import gdcm; print(f"Using gdcm {gdcm.GDCM_VERSION}")'
-        pytest 
+        pytest
+    
+    - name: Install pymedphys
+      if: ${{ matrix.pymedphys-dep == 'pymedphys' }}
+      run: python -m pip install --pre pymedphys[tests]>=0.31.0dev0
+    - name: Get PyMedPhys cache directory
+      if: ${{ matrix.pymedphys-dep == 'pymedphys' }}    
+      id: pymedphys-cache-location
+      run: |
+        echo "::set-output name=dir::$(python -c 'import pymedphys._config as cf; print(str(cf.get_config_dir()))')"
+    - name: PyMedPhys Cache
+      if: ${{ matrix.pymedphys-dep == 'pymedphys' }}    
+      id: pymedphys-cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.pymedphys-cache-location.outputs.dir }}
+        key: pymedphys-${{ hashFiles('**/hashes.json') }}
+    - name: Download all PyMedPhys data for a consistent cache
+      if: ${{ matrix.pymedphys-dep == 'pymedphys' && steps.pymedphys-cache.outputs.cache-hit != 'true' }}
+      run: |
+        python -c "from pymedphys._data import download; download.download_all()"
```

Comment:
You won't need to do this. Scrap this step entirely. Since pymedphys will only ever download the same DICOM files etc. I only needed this step in my workflow since I wanted to reuse the cache across different workflows that downloaded different files.

So, for you, the cache will be consistent by version as long as you always run the same tests. Which you are.